lahja/endpoint.py

```diff
@@ -100,26 +150,45 @@ def connect_no_wait(self, loop: Optional[asyncio.AbstractEventLoop] = None) -> N
 
         self._running = True
 
-    async def connect(self, loop: Optional[asyncio.AbstractEventLoop] = None) -> None:
-        self.connect_no_wait(loop)
-        await self.wait_for_connection()
+    async def start_serving(self,
+                            connection_config: ConnectionConfig,
+                            loop: Optional[asyncio.AbstractEventLoop] = None) -> None:
+        """
+        Start serving this :class:`~lahja.endpoint.Endpoint` so that it can receive events. Await
+        until the :class:`~lahja.endpoint.Endpoint` is ready.
+        """
+        self.start_serving_nowait(connection_config, loop)
+        await self.wait_until_serving()
 
-    async def wait_for_connection(self) -> None:
+    async def wait_until_serving(self) -> None:
         """
-        Wait until the ``Endpoint`` has established a connection to the ``EventBus``
+        Await until the ``Endpoint`` is ready to receive events.
         """
         await asyncio.gather(
```

Comment:
nitpick: I think that the UX has gotten slightly worse here. Previously this method would raise `NoConnection` if this method was called too soon. Now it'll throw an `AttributeError`, because `_receiving_loop_running` and `_internal_loop_running` don't exist yet.

Reply:
Yep, I noticed that, too. I will address this before merging.